docs/HACKING.md

```diff
@@ -0,0 +1,192 @@
+CodeBase Workflow
+========
+
+Report on the working of the code base. It contains the details of file execution and feature implementations in stages. 
+
+## First Stage: Command line execution
+
+  * To format a file user use the command line:
+
+    * `lua-format {lua script} {options}`
+
+    * Where in place of options users can use help, verbose, inplace, dump-config, config, etc.
+
+  * Corresponding to the options given in the command line, their function occurred as specified in the main.cpp.
+
+  * Example:- For help options if user gives either -h or --help
+
+    * `args::HelpFlag help(parser, "help", "Display this help menu", {'h', "help"});`
+
+    * Class help flag shows an instance of help as ‘h’ or  ‘help’ matched.
+
+  * Similarly, with rest option, In ‘config’ or ‘c’ option, `args::ValueFlag<string> cFile`
+
+    * This will read the argument and take the name value of the config file.
+
+* Now read the ConfiFileName by `args::get(cFile);`
+
+* After that check whether to use default config or the configfile given by the user.
```

Comment:
```suggestion
In case it doesn't find any suitable file it will fallback to the default values.

The default values are hard coded in the program (`Config.cpp`) and are used as fallback whenever possible. For instance when fields are not specified in the user's configuration file.
```